### ISSUE TITLE
Ensure RTL domain names are handled correctly

### ISSUE THREAD

**Issue by rekmarks** [MetaMask/metamask-extension#8417]
Inspired by this discussion: https://github.com/MetaMask/metamask-extension/pull/8409#discussion_r415918383

As of today, when we encounter long URLs in the Connected Sites component, we switch the CSS `direction` to `rtl` and use `text-overflow: ellipsis;`. The point is to preserve the TLD in the displayed URL, per [Google's recommendation](https://chromium.googlesource.com/chromium/src/+/master/docs/security/url_display_guidelines/url_display_guidelines.md#eliding-urls).

@Gudahtt raised the point that this may have the opposite affect for domain names with RTL characters in them.

We should review whether we are handled RTL domain names correctly for Connected Sites and apply that handling everywhere.

Some resources:
- https://ux.stackexchange.com/questions/55127/do-arabic-web-browsers-have-the-back-forward-buttons-reversed
- https://bugzilla.mozilla.org/show_bug.cgi?id=525831
- https://tools.ietf.org/html/rfc5893

**Comment by Gudahtt** [comment]
>@Gudahtt raised the point that this may have the opposite affect for domain names with RTL characters in them.

I actually reached the opposite conclusion; that the way we're handling this does seem to be correct.  I believe that URLs are always supposed to be shown left-to-right, even in right-to-left locales. The individual pieces of the domain (i.e. the labels, the period-delimited chunks) should be right-to-left internally when appropriate, but the URL itself is always left-to-right. Which is consistent with what we're doing as far as I know?

**Comment by Gudahtt** [comment]
I see; I missed some of the discussion in that old bugzilla thread about displaying the URL as RTL.

I think we can for the moment dismiss that option, and always render it as LTR. That is the recommendation of most RTL guides I've found (for example, [the MDN guide](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/RTL_Guidelines)), and it's the simplest option for us.

**Comment by rekmarks** [comment]
Yeah, I'm fine with the state of things, but I'll leave this issue open for posterity.
